src/transactions/mod.rs

```diff
@@ -477,6 +488,69 @@ impl<T: inner_mut::PrivateInnerMut + fmt::Debug + Clone + PartialEq> RevaultTran
     }
 }
 
+/// A transaction that can be CPFPed
+pub trait CpfpableTransaction: RevaultTransaction {
+    /// Returns a CpfpTransaction fee-bumping this tx. The package feerate
+    /// of the txs + Cpfp will be increased of `added_feerate` sats/kWU.
+    fn cpfp_transactions(
+        to_be_cpfped: &[(Self, DerivedCpfpDescriptor)],
+        feerate: u64,
+        additional_utxos: Vec<CpfpTxIn>,
+    ) -> Result<CpfpTransaction, TransactionCreationError> {
+        CpfpTransaction::from_txs(to_be_cpfped, feerate, additional_utxos)
+    }
```

Comment:
> This would force us to use different CPFP transactions for the Spends and the Unvaults

Uh, I think that having `to_be_cpfp` as a Vec with `impl CpfpableTransaction` (maybe Boxed?) instead of `Self` could solve. But your fix in the other PR is fine anyways

Reply:
I thought you tried that already :). You can't do that since a `CpfpableTransaction` (requiring `RevaultTransaction`) can't be made into a trait object. https://mitaa.github.io/rust/doc/error-index.html#E0038

Reply:
I tried so many things at this point 😂